python_modules/dagster/dagster/_core/execution/plan/active.py

```diff
@@ -247,26 +270,39 @@ def _update(self) -> None:
             self._executable.append(key)
             del self._waiting_to_retry[key]
 
+        ready_to_claim = []
+        for key in self._waiting_to_claim.keys():
+            if tick_time >= self._waiting_to_claim[key]:
+                ready_to_claim.append(key)
+
+        for key in ready_to_claim:
+            del self._waiting_to_claim[key]
+
     def sleep_til_ready(self) -> None:
         now = time.time()
-        sleep_amt = min([ready_at - now for ready_at in self._waiting_to_retry.values()])
+        ready_at_times = []
+        if self._waiting_to_retry:
+            ready_at_times.extend(self._waiting_to_retry.values())
+        if self._waiting_to_claim:
+            ready_at_times.extend(self._waiting_to_claim.values())
+        sleep_amt = min([ready_at - now for ready_at in ready_at_times])
```

Comment:
i think this looks right but I think its worth trying to explicitly have a test with a step in both retry delay and claim delay